src/views/donate-screen/index.js

```diff
@@ -1,24 +1,85 @@
 import { h } from "preact";
+import { useState, useEffect } from "preact/hooks";
 
 import Title from "../../components/title";
 import Button from "../../components/button";
-import { Content, Subtitle, Container, ContentWrapper } from "./styled";
 
-const DonateScreen = ({ onRequestClose, onNextClick, title }) => (
-  <Container>
-    <Title title={title} onRequestClose={onRequestClose} />
+import {
+  TagItem,
+  Content,
+  Subtitle,
+  Container,
+  CustomInput,
+  TagsWrapper,
+  ContentWrapper,
+} from "./styled";
 
-    <ContentWrapper>
-      <Subtitle>How much would you like to contribute?</Subtitle>
+const DONATION_AMOUNTS = [
+  {
+    text: "1€",
+    value: "1",
+  },
+  {
+    text: "2€",
+    value: "2",
+  },
+  {
+    text: "5€",
+    value: "5",
+  },
+  {
+    text: "10€",
+    value: "10",
+  },
+];
 
-      <Content>
-        donation amount tags comes here and custom input field comes here
-      </Content>
-    </ContentWrapper>
+const DonateScreen = ({ onRequestClose, onNextClick, title }) => {
+  const [inputAmount, setInputAmount] = useState("");
+  const [donationAmount, setDonationAmount] = useState("");
 
-    <Button buttonClick={onNextClick} btnText="Next" />
-  </Container>
-);
+  const handleCustomInput = (e) => {
+    setInputAmount(e.target.value ? e.target.value : "");
+  };
+
+  const onTagSelect = (val) => setDonationAmount(val);
+
+  const handleNextClick = () => {
+    const value = parseInt(donationAmount || inputAmount, 10);
+    onNextClick(value);
+  };
+
+  return (
+    <Container>
+      <Title title={title} onRequestClose={onRequestClose} />
```

Comment:
can we move the this title bar out of each screen and have that in the modal component? 
It is the same for all screens. Then each view is only the relevant view content and can potentially be easier re-used (e.g. for the embedded/non-modal version)

Reply:
yeah, we discussed this in the call yesterday as well. I went with the initial implementation because I need to think on where to place it. And yes it totally makes sense not to have it in each screen. Need to fix the layout as well.

So, I will be doing this later (as a part of the refactoring) if that's fine with you.

Reply:
:+1: yep